File: src/main/webapp/src/model/gallery/galleryItemDto.ts
export interface GalleryItem {
  title?: string
  pictureUrl?: string
  type?: string
  args?: Map<string, string>
}

export class GalleryItemDto implements GalleryItem {
  constructor (
    public title?: string,
    public pictureUrl?: string,
    public type?: string,
    public args?: Map<string, string>
  ) {
  }
}

File: src/main/java/ru/mrak/service/dto/gallery/GalleryItemDto.java
package ru.mrak.service.dto.gallery;

import lombok.Data;
import lombok.experimental.Accessors;
import ru.mrak.model.enumeration.GalleryItemTypeEnum;

import java.util.HashMap;
import java.util.Map;

@Data
@Accessors(chain = true)
public class GalleryItemDto {
    private String title;
    private String pictureUrl;
    private GalleryItemTypeEnum type;
    private Map<String, String> args = new HashMap<>();
}
